### PR TITLE
MHV-64150 MR: Update URL for imaging APIs to remove PII

### PR DESCRIPTION
**Pull request by oleksii-morgun**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- The two controller actions are:
  - ImagingController.request_download
  - ImagingController.images
- Update URL for imaging APIs to remove PII. We have two API endpoints that take "studyId" as a parameter. Study ID is a universal ID and therefore constitutes PII, so we need to remove it from our URLs.
- on `list_imaging_studies` need to ensure that `studyId` values are transformed to `uuid`, mapped and stored in cache so that `request_study` and `list_images` can do a lookup by uuid in cached map, retrieve a real `studyId` value and send request to the upstream service

## Related issue(s)

- [MHV-64150](https://jira.devops.va.gov/browse/MHV-64150)

image

## Testing done

- [x] *New code is covered by unit tests*
- Prior to this change, `/my_health/v1/medical_records/imaging` would include an actual value of `studyIdUrn` in a response which is considered PII. Upon getting response from fire, this value is swapped with random `uuid` and mapped and stored in cache. `/v1/medical_records/imaging/:studyId/request` and `/v1/medical_records/imaging/:studyId/images` would now accept the `uuid` and the client method would locate the right studyId in a cached map before sending a call to FHIR resource.
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
My Health - Medical Records
## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: modules/my_health/docs/openapi.yaml [hunk 1]
```diff
@@ -171,6 +171,111 @@ paths:
...
+      operationId: list_imaging_studies
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                type: object
+                description: >
+                  The schema can be found at
+                  <a target="_blank" href=''>Imaging Studies</a>.
```
Comment (oleksii-morgun) [inline modules/my_health/docs/openapi.yaml:186]: this link here will need to be updated in a separate ticket per @mmoyer-va

File: modules/my_health/docs/openapi.yaml [hunk 2]
```diff
@@ -171,6 +171,111 @@ paths:
...
+      operationId: request_study
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                type: object
+                description: >
+                  The schema can be found at
+                  <a target="_blank" href=''>Imaging Studies</a>.
```
Comment (oleksii-morgun) [inline modules/my_health/docs/openapi.yaml:213]: this link here will need to be updated in a separate ticket per @mmoyer-va

File: modules/my_health/docs/openapi.yaml [hunk 3]
```diff
@@ -171,6 +171,111 @@ paths:
...
+      operationId: list_images
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                type: object
+                description: >
+                  The schema can be found at
+                  <a target="_blank" href=''>Imaging Studies</a>.
```
Comment (oleksii-morgun) [inline modules/my_health/docs/openapi.yaml:252]: this link here will need to be updated in a separate ticket per @mmoyer-va

File: modules/my_health/docs/openapi.yaml (hunk 1 above)
Comment (mmoyer-va) [reply modules/my_health/docs/openapi.yaml:186]: The imaging API documentation is covered by ticket https://jira.devops.va.gov/browse/MHV-63516

File: modules/my_health/docs/openapi.yaml (hunk 2 above)
Comment (mmoyer-va) [reply modules/my_health/docs/openapi.yaml:213]: The imaging API documentation is covered by ticket https://jira.devops.va.gov/browse/MHV-63516

File: modules/my_health/docs/openapi.yaml (hunk 3 above)
Comment (mmoyer-va) [reply modules/my_health/docs/openapi.yaml:252]: The imaging API documentation is covered by ticket https://jira.devops.va.gov/browse/MHV-63516

**Review by mmoyer-va — APPROVED** [review]
This looks good! Tested and works as desired. Nice modification to the specs to set up the redis map there.

**Pull request merged**